libs/hdf-converters/src/conveyor-mapper.ts

```diff
@@ -0,0 +1,244 @@
+import {ExecJSON} from 'inspecjs';
+import _ from 'lodash';
+import {version as HeimdallToolsVersion} from '../package.json';
+import {BaseConverter, ILookupPath, MappedTransform} from './base-converter';
+const DEFAULT_NIST_TAG = ['SI-2', 'RA-5'];
+
+function createDescription(
+  data: Record<string, unknown>,
+  score: number,
+  date: string,
+  type: string
+): Record<string, unknown> {
+  let desc = '';
+  if (type == 'Moldy') {
+    desc =
+      'body:' +
+      _.get(data, 'body') +
+      '\nbody_format:' +
+      (_.get(data, 'body_format') as string) +
+      '\nclassificaton:' +
+      (_.get(data, 'classification') as string) +
+      '\n';
+  } else if (type == 'Stigma') {
+    desc =
+      'body:' +
+      _.get(data, 'body') +
+      '\nbody_format:' +
+      (_.get(data, 'body_format') as string) +
+      '\ntitle_text:' +
+      (_.get(data, 'title_text') as string) +
+      '\nclassificaton:' +
+      (_.get(data, 'classification') as string) +
+      '\n';
+  } else if (type == 'CodeQuality') {
+    desc =
+      'body:' +
+      _.get(data, 'body') +
+      '\nbody_format:' +
+      (_.get(data, 'body_format') as string) +
+      '\nclassificaton:' +
+      (_.get(data, 'classification') as string) +
+      '\ndepth:' +
+      (_.get(data, 'depth') as string) +
+      '\nheuristic:' +
+      (_.get(data, 'heuristic') as string) +
+      '\ntitle_text:' +
+      (_.get(data, 'title_text') as string) +
+      '\n';
+  } else {
+    desc = JSON.stringify(data);
+  }
+  return {
+    status: determineStatus(score),
+    code_desc: desc,
+    start_time: date
+  };
+}
+
+function childrenfinder(output: Record<string, unknown>): string[][] {
+  const arr: string[][] = [];
+  _.forEach(output, (value, key) => {
+    if (_.has(value, 'name')) {
+      const temp: string = _.get(value, 'name[0]') || '';
+      arr.push([key, temp]);
+    }
+    if (_.has(value, 'children')) {
+      const temp: Record<string, unknown> = _.get(value, 'children') || {};
+      _.forEach(childrenfinder(temp), function (value) {
+        arr.push(value);
+      });
+    }
+  });
+  return arr;
+}
+function shafileMapper(
+  output: Record<string, unknown>
+): Record<string, unknown> {
+  const toplevel = _.get(output, 'api_response.file_tree') as Record<
+    string,
+    unknown
+  >;
+  const shamappings = {};
+  _.forEach(toplevel, (value) => {
```

Comment:
this top level code and the helper function seem pretty similar in structure.  this is just based off of a skim so it might not be possible, but it seems like it could be possible to just call the helper function directly from here instead of doing the first iteration in here and then having that iteration call the helper function.

Reply:
fixe

Reply:
fixed